src/java.management/share/classes/javax/management/BinaryRelQueryExp.java

```diff
@@ -177,7 +177,7 @@ public boolean apply(ObjectName name) throws BadStringOperationException, BadBin
             case Query.LE:
                 return sval1.compareTo(sval2) <= 0;
             case Query.EQ:
-                return sval1.compareTo(sval2) == 0;
+                return sval1.equals(sval2);
```

Comment:
I'm not so sure I agree this is better. Using `compareTo` keeps it consistent with the surrounding code.

Reply:
But it's now consistent with Query operator: **EQ**uals

Reply:
All of the other case statements map a Query operator to an arithmetic operator, including mapping EQ to ==, even after your changes (see the previous switch blocks). So your use of `equals` really sticks out as not being consistent with the surrounding code.

Reply:
I agree with Chris on this.

Reply:
I look at it this way, the code should really (IMHO) have been written as:
```
int res = sval1.compareTo(sval2);
switch(relOp) {
case Query.GT: return res > 0;
case ...
}
 and so the use of equals() instead would never have arisen. :)
```

Reply:
Yep, that works for me.

Reply:
Ok. Done.